python/paddle/fluid/tests/unittests/test_detach.py

```diff
@@ -0,0 +1,180 @@
+                    num_flatten_dims=1,
+                    param_attr=fc_w_param_attrs,
+                    bias_attr=fc_b_param_attrs)
+            fc1_w_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(7.0))
+            fc1_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(8.0))
+            fc1 = FC("fc",
+                     1,
+                     num_flatten_dims=1,
+                     param_attr=fc1_w_param_attrs,
+                     bias_attr=fc1_b_param_attrs)
+            data = to_variable(data)
+            conv = fc(data)
+            conv1 = fc1(conv)
+            loss = conv1
+            # print(loss, loss.shape)
+            loss.backward()
+            # print(conv.gradient())
+            return conv.gradient()
+
+    def detach_multi(self):
+        data = self.generate_Data()
+        with fluid.dygraph.guard():
+            fc_w_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(5.0))
+            fc_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(6.0))
+            fc = FC("fc",
+                    10,
+                    num_flatten_dims=1,
+                    param_attr=fc_w_param_attrs,
+                    bias_attr=fc_b_param_attrs)
+            fc1_w_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(7.0))
+            fc1_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(8.0))
+            fc1 = FC("fc",
+                     1,
+                     num_flatten_dims=1,
+                     param_attr=fc1_w_param_attrs,
+                     bias_attr=fc1_b_param_attrs)
+            fc2_w_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(9.0))
+            fc2_b_param_attrs = fluid.ParamAttr(
+                initializer=fluid.initializer.Constant(10.0))
+            fc2 = FC("fc",
+                     1,
+                     num_flatten_dims=1,
+                     param_attr=fc2_w_param_attrs,
+                     bias_attr=fc2_b_param_attrs)
+            data = to_variable(data)
+            conv = fc(data)
+            conv_detach = conv.detach()
+            conv1 = fc1(conv)
+            conv2 = fc2(conv_detach)
+            loss = conv1 + conv2
+            # print(loss, loss.shape)
+            loss.backward()
+            # print(conv.gradient())
+            return conv.gradient()
+
+    def get_Dim(self, array):
+        dim = 1
+        for index_shape in array.shape:
+            dim = dim * index_shape
+        return dim
+
+    def test_NoDetachMulti_DetachMulti(self):
+        array_no_detach_multi = self.no_detach_multi()
+        array_detach_multi = self.detach_multi()
+
+        array_no_detach_multi_dim = self.get_Dim(array_no_detach_multi)
+
+        array_no_detach_multi = array_no_detach_multi.reshape(
```

Comment:
no need to do this, why not get numpy value and use np.array_equalI() to assert the value equality

Reply:
done